### ISSUE TITLE
Create MySQL user for sub servers

### ISSUE THREAD

**Issue by UnleavenedEagle** [virtualmin/virtualmin-gpl#135]
You should add the ability to automatically create a MySQL user for a sub server when creating a MySQL database.

If I have a domain `example.com` and create a top level server it will generate a database `example.com` and a user `example.com` - all with the same name of course.

Now when I generate a sub level server for `sub.example.com`, it will create a database with the name `sub.example.com`, but currently it won't create a database user with the same name, in fact it does not create a user at all.

Now if I want a new user for my sub server, I can create it on my own from Virtualmin, but it's name can only be `sub.sub.example.com` or `*.sub.example.com`. This is quite ugly, especially because this doesn't happen to a top level domain, but this not only creates problems with inconsistency, on some systems due to limited MySQL username length, it can also make some not even that long usernames cut off. This is a major problem.

**Comment by chris001** [comment]
Agreed.  You should have a consistent pattern for the database username/password experience, in sub servers, as in top level virtual servers.

**Comment by jcameron** [comment]
This is by design - sub-servers are under the same ownership as parent servers, so they don't get their own SSH/FTP login, Virtualmin login or MySQL login. However, you can create a user who has access to just the sub-server MySQL DB on the "Edit Users" page.

**Comment by UnleavenedEagle** [comment]
The only problem with that is that if I create a new user for the sub-server, the username will look like sub@sub.example.com or in case of MySQL sub.sub.example.com, while the MySQL database will look like sub_example_com.

If you don't want to add an option for sub-servers to get their own login, I recommend you should add a way to add a "default user" (with the username sub.example.com just like top-servers) to the Edit Users page.

**Comment by jcameron** [comment]
I'll put this on our long-term TODO list..

**Comment by iliaross** [comment]
❗️ If the issue isn't resolved, feel free to reopen it with updated details.
